woocommerce/woocommerce-ios
# Show error when role eligibility request fails

Part of: #7786

### Description
We don't show any error when the role eligibility request fails with an error different from `RoleEligibilityError.insufficientRole`.

This PR aims to show the following error alert for all error scenarios to keep the user informed about the error and provide other options such as contacting support. 

<img src="https://user-images.githubusercontent.com/524475/193964220-22968497-0bbb-49ba-b807-c2e1cb528e9a.png" width="300"/>



### Testing instructions
1. Install and launch the app.
1. Tap `Skip` to reach the login prologue screen.
1. Tap the `Continue with WordPress.com` button and sign in.
1. You will land on the store picker screen from where you can select a store.
1. Turn off the internet to simulate an error.
1. Tap the `Continue` button and verify that you see an error alert. 

### Screenshots
<img src="https://user-images.githubusercontent.com/524475/193964220-22968497-0bbb-49ba-b807-c2e1cb528e9a.png" width="300"/>


---
